pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -221,9 +221,25 @@ func (ctrl *Controller) cascadeDelete(cfg *mcfgv1.KubeletConfig) error {
 	if len(cfg.GetFinalizers()) == 0 {
 		return nil
 	}
-	mcName := cfg.GetFinalizers()[0]
-	err := ctrl.client.MachineconfigurationV1().MachineConfigs().Delete(mcName, &metav1.DeleteOptions{})
+	finalizerName := cfg.GetFinalizers()[0]
+	// for backwards compatibility try deleting the full Machine Config by name
```

Comment:
nit: Won't this continue to be the case once UUIDs start being used _as_ the names themselves?

Reply:
... yes... I force pushed a change to simplify the logic to list the machine configs and then check for ownership for the delete.